include/genn/genn/code_generator/backendBase.h

```diff
@@ -326,6 +326,15 @@ class GENN_EXPORT BackendBase
     //! Get hash digest of this backends identification and the preferences it has been configured with
     virtual boost::uuids::detail::sha1::digest_type getHashDigest() const = 0;
 
+    //! Check if NVTX markers are enabled for this backend
+    virtual bool isNVTXMarkerEnabled() const { return false; }
+    
+    //! Push a named NVTX range marker for profiling
+    virtual void pushNVTXRange(const std::string &) const {}
+    
+    //! Pop the current NVTX range marker
+    virtual void popNVTXRange() const {}
+
```

Comment:
I think you are right that this needs adding to the ``BackendBase`` interface _but_ NVTX is a CUDA-specific technology. I think the virtuals you should define are:
```c++
virtual void genPushProfilerRange(CodeStream &os, const std::string &name) const {}
virtual void genPopProfilerRange(CodeStream &os) const {}
```
which, in the CUDA backend, will check the preferences flag and then write ``nvtxRangePushA(name)`` to ``os`` (I don't see you need to expose the test for whether profiling ranges are enabled)